src/ff/prime_field.rs

```diff
@@ -13,6 +13,7 @@ macro_rules! field_impl {
             const PRIME: Self::Integer = $prime;
             const ZERO: Self = $field(0);
             const ONE: Self = $field(1);
+            const TYPE_STR: &'static str = $type_str;
```

Comment:
Could you instead use `stringify!($field)`?  Then you don't need to worry so much about repeating the name in the invocation.

Reply:
is there a way to also lowercase and/or snake_case it as well?

Reply:
`paste` crate can do that but I am not aware of any std macro that can do this. It may not be that big of a deal though - this string just needs to be unique and we can do case-insensitive comparison using regex if we want to

Reply:
i ended up using `eq_ignore_ascii_case` which seems fine